crates/core/src/rpc/accounts_data.rs

```diff
@@ -637,4 +708,254 @@ mod tests {
             }
         );
     }
+
+    #[ignore = "requires-network"]
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_get_token_supply_with_real_mint() {
```

Comment:
For this test and `test_get_token_supply_caches_remote_account`, can we instead write some dummy account data to the accounts store locally, then verify the RPC returns that data? It would be great to have a baseline version of this test runnable in CI, then we can add tests to fetch from the remote in the integration tests down the road.